sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1152,7 +1154,8 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
         (c, evPrim, evNull) => {
           code"$evPrim = UTF8String.fromString($udtRef.deserialize($c).toString());"
         }
-      case i @ (YearMonthIntervalType | DayTimeIntervalType) =>
+      // TODO(SPARK-XXXXX): Take into account day-time interval fields in cast
```

Comment:
@cloud-fan I have marked the placed where need some follow ups. I am going to create sub-tasks in JIRA.